typing/typecore.ml

```diff
@@ -763,10 +800,8 @@ end) = struct
           if not principal then warn_non_principal warn lid;
           lbl
         | exception Not_found ->
-        match lbls with
-        | (Error(loc', env', err) : _ result) ->
-            Env.lookup_error loc' env' err
-        | Ok lbls ->
+        match filter (force candidates_in_scope) with
+        | Ok lbls | Error lbls ->
```

Comment:
There is a hidden bugfix here. Previously, `lbls` could be empty here, for instance
```ocaml
module M = struct
  type t = { x:int; y:int}
end
type u = { a:int }
let _ = ( { M.x=0 } : u )
```
which triggers an assert false in `Printtyp`.
With this PR this is not the case anymore because we don't fold the filter error case into `Ok [ ]` .

Reply:
I added this example to the testsuite.

Reply:
Note: I cannot reproduce the printing failure with 4.10, this is trunk-only. I wonder if it is a result of the previous refactoring PRs for NameChoice.

Reply:
The bug is present in 4.10, but it requires `-principal` and a non-principal expected type:
```ocaml
module M = struct
  type t = { x:int; y:int}
end
type u = { a:int }
let f (x:u) = x = { M.x=0 }
```